src/promptflow/promptflow/_utils/load_data.py

```diff
@@ -28,21 +22,28 @@ def _pd_read_file(local_path: str, logger: logging.Logger = None) -> pd.DataFram
     ):  # CodeQL [SM01305] Safe use per local_path is set by PRT service not by end user
         return pd.DataFrame()
     # load different file formats
+
+    # set dtype to object to avoid auto type conversion
+    # executor will apply type conversion based on flow definition, so no conversion should be acceptable
+    # note that for csv and tsv format, this will make integer and float columns to be string;
+    # for rest, integer will be int and float will be float
+    dtype = object
+
     if local_path.endswith(".csv"):
-        df = pd.read_csv(local_path, keep_default_na=False)
+        df = pd.read_csv(local_path, dtype=dtype, keep_default_na=False)
     elif local_path.endswith(".json"):
-        df = pd.read_json(local_path)
+        df = pd.read_json(local_path, dtype=dtype)
     elif local_path.endswith(".jsonl"):
-        df = pd.read_json(local_path, lines=True)
+        df = pd.read_json(local_path, dtype=dtype, lines=True)
```

Comment:
Not sure if this kind of dealing could support unlimited rows

Reply:
have offline synced with Executor, previous code does not handle this case, and there will be a design refactor related to this, so we can ignore this issue in this PR.